tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTask.java

```diff
@@ -1022,6 +1023,59 @@ private static Map<String, Object> makeResult(Object data) {
         return m;
     }
 
+    private static GitHubClient createClient(String url) {
+        String host;
+        try {
+            host = (new URL(url)).getHost();
+            if ("github.com".equals(host) || "gist.github.com".equals(host)) {
+                host = "api.github.com";
+            }
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e);
+        }
+
+        return new GitHubClient(host) {
+            @Override
+            protected IOException createException(InputStream response, int code, String status) {
+                String responseBody = null;
+
+                if (this.isError(code)) {
+                    RequestError error;
+                    try {
+                        error = this.parseError(response);
+                    } catch (IOException e) {
+                        return e;
+                    }
+
+                    if (error != null) {
+                        return new RequestException(error, code);
+                    }
+                } else {
+                    try {
+                        responseBody = new String(response.readAllBytes());
+
+                        response.close();
```

Comment:
Move to `finally`?

Reply:
Gave it the ol' try-with-resources treatment.

Reply:
it was copy-paste from original jgit(?) code :)